include/deepworks/image_reader.hpp

```diff
@@ -1,13 +1,16 @@
 #pragma once
 
 #include <string>
+#include <optional>
 
 namespace deepworks {
 
 class Tensor;
 
 namespace io {
 deepworks::Tensor ReadImage(std::string_view);
+
+deepworks::Tensor ReadImageToTensor(std::string_view path, std::optional<Tensor> = std::nullopt);
```

Comment:
Oh, looks hard to understand.
* What's the purpose not provide `Tensor` to this function, if it's possible to use `ReadImage` instead
* I believe it just `void ReadImageToTensor(std::string_view path, Tensor& out)`